corehq/apps/cloudcare/static/cloudcare/js/formplayer/menus/views.js

```diff
@@ -1687,6 +1696,7 @@ hqDefine("cloudcare/js/formplayer/menus/views", [
             const smallScreenEnabledOnStartup = cloudcareUtils.smallScreenIsEnabled();
             const arrowToggle = $('#persistent-menu-arrow-toggle');
             self.makeCollapse(sessionStorage.showPersistentMenu);
+            self.menuCollapseExpandTransitionListener();
```

Comment:
Is `onAttach` called everytime this view is rendered? In which case, would create an eventListener everytime and lead to multiple being created?

Reply:
`onAttach` is called every time the view is rendered, but the end of the life is `onDestroy`, The listener should destroyed along with the element it's attached to when the view is destroyed and re-rendered. I did double check to be sure and there aren't multiple listeners being created.

Reply:
interesting, that's nice to learn. I didn't know they interacted that way. Thanks for checking!